From 29f271176982a2b5c20a7fafd93befaace2ea650 Mon Sep 17 00:00:00 2001
From: Peter Swain <pswain@cavium.com>
Date: Fri, 13 Feb 2015 13:58:13 +0530
Subject: [PATCH 037/132] usb: dwc3: simplify OCTEON ifdefs

Commit c903cd656ec7ea279cd9856278c7fde4e5656391 from
git://git.yoctoproject.org/linux-yocto-3.14

Signed-off-by: Peter Swain <pswain@cavium.com>
Signed-off-by: Abhishek Paliwal <abhishek.paliwal@aricent.com>
[Original patch taken from OCTEON-SDK 3.1.1-544.
 Fix some compile warnings.]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/usb/dwc3/core.c | 28 +++++++++++++++-------------
 1 file changed, 15 insertions(+), 13 deletions(-)

diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index f802ee9..9ed5325e 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -50,7 +50,7 @@
 extern void octeon3_usb_phy_reset(int index);
 extern int xhci_octeon_start(struct platform_device *pdev);
 extern int xhci_octeon_stop(struct platform_device *pdev);
-#ifdef CONFIG_USB_XHCI_HCD_OCTEON
+#if IS_ENABLED(CONFIG_USB_XHCI_HCD_OCTEON)
 static u64 xhci_octeon_dma_mask = DMA_BIT_MASK(64);
 #endif
 
@@ -74,7 +74,7 @@ void dwc3_set_mode(struct dwc3 *dwc, u32 mode)
 static int dwc3_core_soft_reset(struct dwc3 *dwc)
 {
 	u32		reg;
-	int		ret;
+	int		ret = 0;
 	int		index;
 
 	/* Before Resetting PHY, put Core in Reset */
@@ -93,7 +93,7 @@ static int dwc3_core_soft_reset(struct dwc3 *dwc)
 	dwc3_writel(dwc->regs, DWC3_GUSB2PHYCFG(0), reg);
 
 	index = (((uint64_t)(dwc->regs) & 0x10000000000ull) ? 1:0);
-#ifdef CONFIG_USB_XHCI_HCD_OCTEON
+#if IS_ENABLED(CONFIG_USB_XHCI_HCD_OCTEON)
 	octeon3_usb_phy_reset(index);
 #else
 	usb_phy_init(dwc->usb2_phy);
@@ -127,7 +127,7 @@ static int dwc3_core_soft_reset(struct dwc3 *dwc)
 	reg &= ~DWC3_GCTL_CORESOFTRESET;
 	dwc3_writel(dwc->regs, DWC3_GCTL, reg);
 
-	return 0;
+	return ret;
 }
 
 /**
@@ -585,7 +585,7 @@ err2:
 	dwc3_free_scratch_buffers(dwc);
 
 err1:
-#ifndef CONFIG_USB_XHCI_HCD_OCTEON
+#if !IS_ENABLED(CONFIG_USB_XHCI_HCD_OCTEON)
 	usb_phy_shutdown(dwc->usb2_phy);
 	usb_phy_shutdown(dwc->usb3_phy);
 	phy_exit(dwc->usb2_generic_phy);
@@ -598,7 +598,7 @@ err0:
 
 static void dwc3_core_exit(struct dwc3 *dwc)
 {
-#ifndef CONFIG_USB_XHCI_HCD_OCTEON
+#if !IS_ENABLED(CONFIG_USB_XHCI_HCD_OCTEON)
 	dwc3_free_scratch_buffers(dwc);
 	usb_phy_shutdown(dwc->usb2_phy);
 	usb_phy_shutdown(dwc->usb3_phy);
@@ -755,7 +755,7 @@ static int dwc3_probe(struct platform_device *pdev)
 	void __iomem		*regs;
 	void			*mem;
 
-#ifdef CONFIG_USB_XHCI_HCD_OCTEON
+#if IS_ENABLED(CONFIG_USB_XHCI_HCD_OCTEON)
 	ret = xhci_octeon_start(pdev);
 	if (ret)
 		return ret;
@@ -907,7 +907,7 @@ static int dwc3_probe(struct platform_device *pdev)
 	platform_set_drvdata(pdev, dwc);
 
 	if (!dev->dma_mask) {
-#ifdef CONFIG_USB_XHCI_HCD_OCTEON
+#if IS_ENABLED(CONFIG_USB_XHCI_HCD_OCTEON)
 		dev->dma_mask = &xhci_octeon_dma_mask;
 #else
 		dev->dma_mask = dev->parent->dma_mask;
@@ -943,7 +943,7 @@ static int dwc3_probe(struct platform_device *pdev)
 		goto err1;
 	}
 
-#ifndef CONFIG_USB_XHCI_HCD_OCTEON
+#if !IS_ENABLED(CONFIG_USB_XHCI_HCD_OCTEON)
 	usb_phy_set_suspend(dwc->usb2_phy, 0);
 	usb_phy_set_suspend(dwc->usb3_phy, 0);
 	ret = phy_power_on(dwc->usb2_generic_phy);
@@ -984,11 +984,13 @@ err5:
 err4:
 	phy_power_off(dwc->usb3_generic_phy);
 
+#if !IS_ENABLED(CONFIG_USB_XHCI_HCD_OCTEON)
 err3:
+#endif
 	phy_power_off(dwc->usb2_generic_phy);
 
+#if !IS_ENABLED(CONFIG_USB_XHCI_HCD_OCTEON)
 err2:
-#ifndef CONFIG_USB_XHCI_HCD_OCTEON
 	usb_phy_set_suspend(dwc->usb2_phy, 1);
 	usb_phy_set_suspend(dwc->usb3_phy, 1);
 #endif
@@ -1025,7 +1027,7 @@ static int dwc3_remove(struct platform_device *pdev)
 	dwc3_event_buffers_cleanup(dwc);
 	dwc3_free_event_buffers(dwc);
 
-#ifndef CONFIG_USB_XHCI_HCD_OCTEON
+#if !IS_ENABLED(CONFIG_USB_XHCI_HCD_OCTEON)
 	usb_phy_set_suspend(dwc->usb2_phy, 1);
 	usb_phy_set_suspend(dwc->usb3_phy, 1);
 	phy_power_off(dwc->usb2_generic_phy);
@@ -1037,7 +1039,7 @@ static int dwc3_remove(struct platform_device *pdev)
 	pm_runtime_put_sync(&pdev->dev);
 	pm_runtime_disable(&pdev->dev);
 
-#ifdef CONFIG_USB_XHCI_HCD_OCTEON
+#if IS_ENABLED(CONFIG_USB_XHCI_HCD_OCTEON)
 	xhci_octeon_stop(pdev);
 #endif
 	return 0;
@@ -1065,7 +1067,7 @@ static int dwc3_suspend(struct device *dev)
 	dwc->gctl = dwc3_readl(dwc->regs, DWC3_GCTL);
 	spin_unlock_irqrestore(&dwc->lock, flags);
 
-#ifndef CONFIG_USB_XHCI_HCD_OCTEON
+#if !IS_ENABLED(CONFIG_USB_XHCI_HCD_OCTEON)
 	usb_phy_shutdown(dwc->usb3_phy);
 	usb_phy_shutdown(dwc->usb2_phy);
 	phy_exit(dwc->usb2_generic_phy);
-- 
1.9.1

